From abc88d52ee5e86decd5e98b3223d3feec0dd66bc Mon Sep 17 00:00:00 2001
From: xtex <xtexchooser@duck.com>
Date: Sat, 7 Dec 2024 21:12:25 +0800
Subject: [PATCH 1/3] Fix symbolic links for Linux kernel 6.13-rc1
X-Developer-Signature: v=1; a=openpgp-sha256; l=2446; i=xtexchooser@duck.com;
 h=from:subject; bh=4orMdTlyFiafvwXo+YIZ8q0Emww7pH1RBNOMMhTxeIA=;
 b=owGbwMvMwCW2U4Ij7wZL9ETG02pJDOkhRf8mHj1+2F2S4/++tk08GT2s/1V53Oae4X80O37Fm
 Z7LB/Tnd5SyMIhxMciKKbIUGTZ4s+qk84suK5eFmcPKBDKEgYtTACby15Lhf3xXOu/XpxkPLX0y
 PrS79T/1r0twMuGZtCmd8WMUF+fFtwx/pbrnVs2cIGt6WOZU9OU74Uq3f27f8yCoTNHmfW9mpaY
 hLwA=
X-Developer-Key: i=xtexchooser@duck.com; a=openpgp;
 fpr=7231804B052C670F15A6771DB918086ED8045B91

Link: https://gist.github.com/joanbm/d1f89391a4b20f4b56ba931ef6ca62da
Link: https://github.com/NVIDIA/open-gpu-kernel-modules/issues/747
Author: Joan Bruguera <joanbrugueram@gmail.com>
---
 nvidia-modeset/nvidia-modeset.Kbuild | 8 +++++---
 nvidia/nvidia.Kbuild                 | 8 +++++---
 2 files changed, 10 insertions(+), 6 deletions(-)

diff --git a/nvidia-modeset/nvidia-modeset.Kbuild b/nvidia-modeset/nvidia-modeset.Kbuild
index 9698b59e24de..75e3563cde37 100644
--- a/nvidia-modeset/nvidia-modeset.Kbuild
+++ b/nvidia-modeset/nvidia-modeset.Kbuild
@@ -40,13 +40,15 @@ NV_KERNEL_MODULE_TARGETS += $(NVIDIA_MODESET_KO)
 NVIDIA_MODESET_BINARY_OBJECT := $(src)/nvidia-modeset/nv-modeset-kernel.o_binary
 NVIDIA_MODESET_BINARY_OBJECT_O := nvidia-modeset/nv-modeset-kernel.o
 
-quiet_cmd_symlink = SYMLINK $@
-cmd_symlink = ln -sf $< $@
+# Rel. commit 80f289101690 "kbuild: change working directory to external module directory with M=" (Masahiro Yamada, 10 Nov 2024)
+# Ensure `$<` is absolute, since the link target is resolved relative to its path, not from where `ln` is run from.
+quiet_cmd_symlinkabs = SYMLINK $@
+ cmd_symlinkabs = ln -sf $(abspath $<) $@
 
 targets += $(NVIDIA_MODESET_BINARY_OBJECT_O)
 
 $(obj)/$(NVIDIA_MODESET_BINARY_OBJECT_O): $(NVIDIA_MODESET_BINARY_OBJECT) FORCE
-	$(call if_changed,symlink)
+	$(call if_changed,symlinkabs)
 
 nvidia-modeset-y += $(NVIDIA_MODESET_BINARY_OBJECT_O)
 
diff --git a/nvidia/nvidia.Kbuild b/nvidia/nvidia.Kbuild
index ea4ef5badea7..4262c6f35fb3 100644
--- a/nvidia/nvidia.Kbuild
+++ b/nvidia/nvidia.Kbuild
@@ -40,13 +40,15 @@ NVIDIA_KO = nvidia/nvidia.ko
 NVIDIA_BINARY_OBJECT := $(src)/nvidia/nv-kernel.o_binary
 NVIDIA_BINARY_OBJECT_O := nvidia/nv-kernel.o
 
-quiet_cmd_symlink = SYMLINK $@
- cmd_symlink = ln -sf $< $@
+# Rel. commit 80f289101690 "kbuild: change working directory to external module directory with M=" (Masahiro Yamada, 10 Nov 2024)
+# Ensure `$<` is absolute, since the link target is resolved relative to its path, not from where `ln` is run from.
+quiet_cmd_symlinkabs = SYMLINK $@
+ cmd_symlinkabs = ln -sf $(abspath $<) $@
 
 targets += $(NVIDIA_BINARY_OBJECT_O)
 
 $(obj)/$(NVIDIA_BINARY_OBJECT_O): $(NVIDIA_BINARY_OBJECT) FORCE
-	$(call if_changed,symlink)
+	$(call if_changed,symlinkabs)
 
 nvidia-y += $(NVIDIA_BINARY_OBJECT_O)
 

base-commit: 6127483bad1f7ecf052c9aaba198452d4cf47ce9
-- 
2.47.1

